File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    var answer = 0;
    
    const queue = queue1.concat(queue2);
    const total = queue.reduce((acc, cur)=> acc+cur, 0);
    if(total%2!==0) return -1;  // 전체 합이 홀수면 불가능
    
    let l=0, r=queue1.length, target = total/2;
    let currentSum = queue1.reduce((acc, cur)=>acc+cur, 0);
    
    while(r<queue.length){
        if(currentSum===target)    return answer;
        
        else if(currentSum > target){
            currentSum -= queue[l++];
        }
        else if(currentSum < target){
            currentSum += queue[r++];
        }
        
        answer++;
    }
    
    return -1;
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
def solution(queue1, queue2):
    answer = 0
    queue = queue1 + queue2
    total = sum(queue) 
    if total % 2 != 0:
        return -1
    target = total // 2
    
    l = 0
    r = len(queue1) - 1
    cur = sum(queue1)
    
    while r < len(queue):
        if cur == target:
            return answer
        
        if cur > target:
            cur -= queue[l]
            l += 1
        else:
            r += 1
            if r >= len(queue):
                return -1
            cur += queue[r]
        
        answer += 1
        
    return -1